### ISSUE TITLE
Develop a prototype implementation for source-based coverage report generation

### ISSUE THREAD

**Issue by adpaco-aws** [model-checking/kani#43]
Source-based coverage report generation is an experimental feature of the Rust compiler that we need to explore further. A prototype implementation will allow us to discover any limitations that may influence the way we plan to structure the RMC testing suite.

**Comment by celinval** [comment]
https://doc.rust-lang.org/rustc/instrument-coverage.html

**Comment by adpaco-aws** [comment]
Thanks, @celinval! Should try this some time, last time it was still an unstable option 😬

**Comment by adpaco-aws** [comment]
I recall that, when I opened this issue, I wanted to report source-based coverage FOR source code in Kani. In other words, I wanted to use the coverage feature in the Rust compiler to obtain coverage reports for the kani compiler, its driver, etc. whenever we ran our tests, to ensure that we were covering all cases we were expecting to cover.

This shouldn't be very difficult to achieve now that the coverage feature is more mature, but it doesn't seem to be a priority for us. Otherwise, this is essentially a duplicate of #2640. For that reason, I'm closing this issue until we decide to obtain coverage metrics for Kani itself.
